src/Common/src/System/Security/Cryptography/RSACng.SignVerify.cs

```diff
@@ -40,6 +64,11 @@ public override byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RS
 
             using (SafeNCryptKeyHandle keyHandle = GetDuplicatedKeyHandle())
             {
+                if (hash.Length != GetHashSizeInBytes(hashAlgorithm))
```

Comment:
I intentionally had it down this far so that we more closely emulated the behavior of the OS being the thing to reject the value.  That said, I can't think of a reason that this needs to stay this way.  So I'll move it.